pageserver/src/tenant/remote_timeline_client/upload.rs

```diff
@@ -120,17 +119,17 @@ pub(crate) async fn upload_initdb_dir(
     storage: &GenericRemoteStorage,
     tenant_id: &TenantId,
     timeline_id: &TimelineId,
-    initdb_dir: Bytes,
+    initdb_tar_zst: File,
+    size: u64,
 ) -> anyhow::Result<()> {
     tracing::trace!("uploading initdb dir");
 
-    let size = initdb_dir.len();
-
-    let bytes = futures::stream::once(futures::future::ready(Ok(initdb_dir)));
+    let buf_reader = tokio::io::BufReader::new(initdb_tar_zst);
+    let file = tokio_util::io::ReaderStream::with_capacity(buf_reader, 8 * 1024);
```

Comment:
```suggestion
    let file = tokio_util::io::ReaderStream::with_capacity(buf_reader, 32 * 1024);
```

Let's do it on this PR which introduces the file io.

Reply:
done